### PR TITLE
Proxy task 5:

### PR DESCRIPTION
**Pull request by a-prakaptsova**
- structure reorganisation
- endpoint /meteors have been added

### CODE REVIEW

File: src/utils/dateHelper.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const { previousFriday, previousMonday, format } = require("date-fns");
+require('dotenv').config();
+
+const END_DATE_RAW_FORMAT = previousFriday(new Date());
```
Comment (plutsenka) [inline src/utils/dateHelper.js]: It seems like this one is actually END_DATE, not FORMAT

File: src/utils/dateHelper.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+const START_DATE_RAW_FORMAT = previousMonday(END_DATE_RAW_FORMAT);
+
+const START_DATE = format(START_DATE_RAW_FORMAT, process.env.DATE_FORMAT);
```
Comment (plutsenka) [inline src/utils/dateHelper.js]: So this one can be START_DATE_FORMATTED

File: src/services/asteroidService.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+const getAsteroidsInPeriod = () => {
+    return axios.get(process.env.GET_ASTEROIDS_URL, {
+        params: {
+            start_date: dates.START_DATE,
+            end_date: dates.END_DATE,
+            api_key: process.env.API_KEY
+        }
+    })
+        .then((responce) => {
```
Comment (plutsenka) [inline src/services/asteroidService.js]: I believe it is responSe

**Review by plutsenka — APPROVED** [review]

File: src/controllers/asteroidController.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+const getAsteroids = require('../services/asteroidService')
+
+const getAsteroidsInPeriod = (req, res) => {
+    res.contentType('application/json');
+    getAsteroids()
+        .then(data => res.contentType('application/json')
```
Comment (mdziarkach) [inline src/controllers/asteroidController.js]: if you use 'application/json' by default then you can add this header once somewhere in root server.js file

File: src/index.js [hunk 5]
```diff
@@ -1,37 +1,11 @@
-const { previousFriday, previousMonday, format } = require("date-fns");
-const axios = require('axios');
+const express = require('express');
+const app = express();
+const asteroidRouter = require('./routes/asteroidsRouter');
 require('dotenv').config();
 
-const END_DATE = previousFriday(new Date());
-const START_DATE = previousMonday(END_DATE);
+const PORT = process.env.PORT;
```
Comment (mdziarkach) [inline src/index.js]: const PORT = process.env.PORT || 8000;
if .env file wasn't exist

File: src/services/asteroidService.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+const axios = require('axios');
+const consolePrinter = require('../utils/consolePrinter');
+const dates = require('../utils/dateHelper');
+require('dotenv').config();
```
Comment (mdziarkach) [inline src/services/asteroidService.js]: you need to add this line only once in root file

File: src/services/asteroidService.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+const axios = require('axios');
+const consolePrinter = require('../utils/consolePrinter');
```
Comment (mdziarkach) [inline src/services/asteroidService.js]: let's call it logger

File: src/utils/consolePrinter.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+const printAsteroidCountToConsole = (asteroidCount, startDate, endDate) => {
+    console.log(`From ${startDate} to ${endDate} were seen ${asteroidCount} asteroid(s).`);
+}
+
+const printErrorMessage = (error) => {
+    console.error(`Error occurred during request: ${error.message}`);
+}
+
```
Comment (mdziarkach) [inline src/utils/consolePrinter.js]: can you rename functions please? `log`, `warn`, `error`

File: src/index.js [hunk 9]
```diff
@@ -1,37 +1,15 @@
...
+const PORT = process.env.PORT || 8000;
+app.listen(PORT, (error) => {
+    error ? console.error(error) : console.log(`Server is running on the port ${PORT}`);
+});
+
+app.use((req, res, next) => {
+    res.setHeader("Content-Type", "application/json");
+    next();
+});
+app.use('/', asteroidRouter);
```
Comment (mdziarkach) [inline src/index.js]:
```js
this.app.use(exceptionFilter);
this.app.use('*', (req, res) =>
      res.status(404).json({ message: 'Page not found' }),
 );
```

File: src/controllers/asteroidController.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+const getAsteroids = require('../services/asteroidService')
+
+const getAsteroidsInPeriod = (req, res) => {
+    getAsteroids()
+        .then(data => res.status(200)
+                .json(data))
+        .catch(error => res.status(500)
```
Comment (mdziarkach) [inline src/controllers/asteroidController.js]: Each controller must handle possible errors. Over time, there may be dozens and hundreds of such places. It is advisable to report errors in a uniform manner
The easiest way is to handle errors in one place. At the controller level, if an error occurs, you only need to call `next(error)` and handle this error at the very end of the middleware chain
```js
.catch(error => next(error)
```
and then you can catch it in exception filter here `src/index.js`

File: src/index.js [hunk 11]
```diff
@@ -1,37 +1,15 @@
...
-}
-
-getAllAsteroidsInPeriod();
\ No newline at end of file
+const express = require('express');
+const app = express();
+const asteroidRouter = require('./routes/asteroidsRouter');
+
+const PORT = process.env.PORT || 8000;
+app.listen(PORT, (error) => {
```
Comment (mdziarkach) [inline src/index.js]: usually this listener somewhere in the end after all handlers

File: src/services/asteroidService.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    return axios.get(process.env.GET_ASTEROIDS_URL, {
+        params: {
+            start_date: dates.START_DATE,
+            end_date: dates.END_DATE,
+            api_key: process.env.API_KEY
+        }
+    })
+        .then((resp) => {
+            logger.logJson(resp.data);
+            logger.logAsteroidCount(resp.data.element_count, dates.START_DATE, dates.END_DATE);
```
Comment (mdziarkach) [inline src/services/asteroidService.js]: does it means you will create new loggers for each entity?
```js
loger.log(JSON.stringify(resp.data, null, 2))
loger.log(`From ${dates.START_DATE} to ${dates.END_DATE} were seen ${resp.data.element_count} asteroid(s).`)
```

File: src/services/asteroidService.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+            return resp.data;
+        })
+        .catch((err) => {
+            logger.error(err);
```
Comment (mdziarkach) [inline src/services/asteroidService.js]: need to throw exception here and controller should catch it and send it to `next`

File: src/services/asteroidService.js [hunk 14]
```diff
@@ -0,0 +1,24 @@
+const axios = require('axios');
+const logger = require('../utils/logger');
+const dates = require('../utils/dateHelper');
+
+console.log(process.env.DATE_FORMAT);
```
Comment (mdziarkach) [inline src/services/asteroidService.js]: please remove it

Comment (a-prakaptsova) [reply src/services/asteroidService.js]: everything is fixed

**Pull request merged**